Sources/ApolloCodegenLib/Frontend/JavaScript/src/__tests__/deferCompilationTests.ts

```diff
@@ -152,4 +154,46 @@ describe("given schema", () => {
     });
   });
 
+  describe("query has inline fragment with @defer directive and no type condition should throw error", () => {
+    const documentString: string = `
+    query Test {
+      allAnimals {
+        ... @defer {
+          species
+        }
+      }
+    }
+    `;
+
+    const document: DocumentNode = parseOperationDocument(
+      new Source(documentString, "Test Query", { line: 1, column: 1 })
+    );
+
+    it("should compile inline fragment with directive", () => {
+      const validationErrors: readonly GraphQLError[] = validateDocument(schema, document, emptyValidationOptions)
+      expect(validationErrors.length).toEqual(1)
+    });
+  });
+
+  describe("query has inline fragment with @defer directive should not throw error", () => {
```

Comment:
I believe this case is already covered by the test up on [line 37](https://github.com/apollographql/apollo-ios/pull/3159/files#diff-9764f88416cc3dd09bf331172384c2bb39d377cf1547465df39e49755aad50c6R37). Can remove this as a duplicate unless I'm missing something here.